src/foam/nanos/ruler/tracer/LoggingTracer.js

```diff
@@ -55,10 +55,11 @@
         { name: 'oldObj', type: 'foam.core.FObject' }
       ],
       javaCode: `
-        setLog(getLog().append("\\nRule: "));
-        setLog(getLog().append(rule.getName()));
-        setLog(getLog().append(", id: "));
-        setLog(getLog().append(rule.getId()));
+        StringBuilder log = getLog();
+        log.append("\\nRule: "));
+        log.append(rule.getName()));
+        log.append(", id: "));
+        setLog(log.append(rule.getId()));
```

Comment:
Well if you're going to optimize this .. might as well do it right:
```suggestion
        setLog(getLog()
         .append("\\nRule: "))
          .append(rule.getName()))
          .append(", id: "))
          .append(rule.getId()));
```